docs/install/file-storage/README.md

```diff
@@ -0,0 +1,115 @@
+# FlowForge File Storage
+
+When running in cloud like environment (under Docker or Kubernetes)
+FlowForge projects do not have access to a persistent filesystem to 
+use to store files. Files written to the container file system will 
+lost if the project is restarted. 
+
+For this reason we recomend that the Node-RED core file nodes are
+disabled on these platforms.
+
+To solve this limitation FlowForge provides a set of replacement
+file nodes. These are backed by Object Store which can be configured
+to use a mounted volume or S3 bucket.
+
+The FlowForge File Storage system can also be used with the FlowForge
+LocalFS build, but in this case the FlowForge Projects already have
+direct access to the file system of the machine hosting FlowForge.
+
+## Configuring
+
+The Docker Compose and Kubernetes Helm Chart will both now start a 
+container running the File Server application.
+
+There are 2 main things to set up:
+
+ 1. Which backend to use (`localfs` or `s3`)
+ 2. The configuration options for the backend
+
+### LocalFS
+
+This can be used when you mount a volume into the File Server container
+to persist files.
+
+The only configuration option used is the path to the directory to use
+as the root of the storage
+
+```
+host: 0.0.0.0
+port: 3001
+base_url: http://flowforge:3000
+driver:
+  type: localfs
+  options:
+    root: var/root
+```
+
+### S3
+
+The following can be any of the options for the S3Client Contructor, see [here](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-s3/interfaces/s3clientconfig.html)
+
+- options
+    - bucket - name of S3 Bucket (required)
+    - region - AWS Region
+    - endpoint - S3 ObjectStore Endpoint (if not using AWS S3)
+    - forcePathStyle: true/false
+    - credential
+        - accessKeyId - AccountID/Username
+        - secretAccessKey - SecretKey/Password
+
+e.g.
+
+```
+host: '0.0.0.0'
+port: 3001
+base_url: http://forge.default
+driver:
+  type: s3
+  options:
+    bucket: flowforge-files
+    credentials:
+      accessKeyId: XXXXXXXXXXXXXXXXXXX
+      secretAccessKey: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+    forcePathStyle: true
+    region: us-east-1
+```
+
+## Docker Compose
+
+You can configure the backend store for the File Server by editing the 
+`etc/flowforge-storage.yml` file. It defaults to using the `localfs` 
+backend driver and mounting a directory into the contianer.
+
+## Kubernetes Helm
+
+You can configure the backend store for the File server by including 
+the following values passed to helm
+
+- `forge.fileStore.type` - defaults to `s3`
+- `forge.fileStore.options` - an object that matches the Yaml described above
+
+## Enabling the FlowForge File Nodes
+
+The FlowForge File nodes have been written to be direct replacements
+for the Node-RED core file-in and file-out nodes. This means that only 
+one version of these nodes can be active in a project at a time.
+
+The FlowForge File nodes will automatically disable themselves if the 
```

Comment:
When FlowForge creates the template during the install, do users need to go through this? If so that should be documented in ../install/*

Reply:
This is not currently in the auto generated template, but we should add it (but not today)